### PR TITLE
[opentitantool] Allow sending/receiving plain bytes via SPI

### PR DESCRIPTION
**Pull request by jesultra**
Sometimes HyperDebug could be connected to a non-eeprom/flash device, or sometimes the user might want to send some non-standard command to a SPI eeprom/flash.

This CL introduces commands similar to what exists for I2C to send and receive arbitrary byte sequences.

### CODE REVIEW

File: sw/host/opentitantool/src/command/spi.rs [hunk 1]
```diff
@@ -305,6 +302,86 @@ impl CommandDispatch for SpiTpm {
...
+        let context = context.downcast_ref::<SpiCommand>().unwrap();
+        let spi_bus = context.params.create(transport, "BOOTSTRAP")?;
+        spi_bus.run_transaction(&mut [Transfer::Write(&hex::decode(&self.hexdata)?)])?;
+        Ok(None)
+    }
+}
+
+/// Write plain data bytes to a SPI device (not necessarily SPI EEPROM/flash).
+#[derive(Debug, StructOpt)]
+pub struct SpiRawWriteRead {
```
Comment (cfrantz) [inline sw/host/opentitantool/src/command/spi.rs:357]: Is this a Write followed by a Read, or a a simultaneous "Transceive" operation?

Comment (jesultra) [reply sw/host/opentitantool/src/command/spi.rs:357]: The though occurred to me, but then I forgot.  This is write followed by read, maybe it should be called `opentitantool spi raw-write-then-read`.  I can also add a `spi raw-transcieve` for completeness.

**Comment by jesultra** [comment]
I have clarified and added a method for simultaneous "transceive".

File: sw/host/opentitantool/src/command/spi.rs (hunk 1 above)
Comment (cfrantz) [reply sw/host/opentitantool/src/command/spi.rs]: I don't have a problem with either `raw-write-read` or `raw-write-then-read`; I prefer the former for being shorter.  Just needed some clarification on exactly what it was meant to do.

**Review by cfrantz — APPROVED** [review]

File: sw/host/opentitantool/src/command/spi.rs (hunk 1 above)
Comment (jesultra) [reply sw/host/opentitantool/src/command/spi.rs]: Ok, let's go with the shorter names.

**Pull request merged**